rust/headless-client/src/ipc_service.rs

```diff
@@ -540,15 +565,18 @@ impl<'a> Handler<'a> {
         // Call `set_dns` before `set_tun` so that the tunnel starts up with a valid list of resolvers.
         tracing::debug!(?dns, "Calling `set_dns`...");
         connlib.set_dns(dns);
+        let tun_span = transaction.start_child("tun", "Raise tunnel");
         let tun = self
             .tun_device
             .make_tun()
             .map_err(|e| Error::TunnelDevice(e.to_string()))?;
         connlib.set_tun(Box::new(tun));
+        tun_span.finish();
 
         let session = Session { cb_rx, connlib };
```

Comment:
The `Session` object internally has its own panic handler for the connlib task. Until we have `tracing` integration that automatically captures all errors, can we add a manual error submission there any time connlib panics?

Reply:
You could also do it as part of the `on_disconnect` callback although we might be able to capture more details of the error if we do it there directly?